eng/scripts/get-aspire-cli.ps1

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env pwsh
+
+[CmdletBinding()]
+param()
+
+$ErrorActionPreference = 'Stop'
+
+# Define supported combinations (script-level constant)
+$script:SupportedCombinations = @(
+    "win-x86",
+    "win-x64",
+    "win-arm64",
+    "linux-x64",
+    "linux-arm64",
+    "linux-musl-x64",
+    "osx-x64",
+    "osx-arm64"
+)
+
+# Function to detect OS
+function Get-OperatingSystem {
+    if ($IsWindows -or ($PSVersionTable.PSVersion.Major -le 5)) {
+        return "win"
+    }
+    elseif ($IsLinux) {
+        # Check if it's musl-based (Alpine, etc.)
+        try {
+            $lddOutput = & ldd --version 2>&1 | Out-String
+            if ($lddOutput -match "musl") {
+                return "linux-musl"
+            }
+            else {
+                return "linux"
+            }
+        }
+        catch {
+            return "linux"
+        }
+    }
+    elseif ($IsMacOS) {
+        return "osx"
+    }
+    else {
+        return "unsupported"
+    }
+}
+
+# Function to detect architecture
+function Get-Architecture {
+    $arch = [System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture
+
+    switch ($arch) {
+        "X64" { return "x64" }
+        "Arm64" { return "arm64" }
+        "X86" { return "x86" }
+        default { return "unsupported" }
+    }
+}
+
+# Function to validate OS/arch combination
+function Test-SupportedCombination {
+    param(
+        [string]$OS,
+        [string]$Architecture
+    )
+
+    $combination = "$OS-$Architecture"
+
+    return $combination -in $script:SupportedCombinations
+}
+
+# Main function
+function Main {
+    try {
+        # Detect OS and architecture
+        $os = Get-OperatingSystem
+        $arch = Get-Architecture
+
+        # Check for unsupported OS or architecture
+        if ($os -eq "unsupported") {
+            Write-Error "Error: Unsupported operating system: $([System.Environment]::OSVersion.Platform)"
+            exit 1
```

Comment:
isn't using exit 1 inconsistent with the above ErrorActionPreference = Stop? Shouldn't this be throw?